### PR TITLE
Deploy documentation using Github pages

### PR DESCRIPTION
**Pull request by yhy-1**
Switch from netlify to GitHub pages and update some readme.md that was missed.

You can check it publishing to my forked repo gh-pages branch @dgreene1 
https://github.com/yhy-1/react-accessible-treeview/tree/gh-pages

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,7 +10,10 @@ A react component that implements the treeview pattern as described by the [WAI-
 - Highly customizable through the use of the render prop and prop getter patterns.
 - WAI-ARIA compliant.
 
-[Documentation and demo](https://react-accessible-treeview.netlify.com)
+### Documentation and Demo
+
+- https://lissitz.github.io/react-accessible-treeview
```
Comment (dgreene1) [inline README.md]: I think it’s going to be under my link not lissitz

File: website/docusaurus.config.js [hunk 2]
```diff
@@ -9,15 +9,16 @@ module.exports = {
   title: "react-accessible-treeview",
   tagline:
     "A React component that implements the treeview pattern as described by the WAI-ARIA Authoring Practices.",
-  url: "https://react-accessible-treeview.netlify.com",
-  baseUrl: "/",
+  url: "https://lissitz.github.io",
```
Comment (dgreene1) [inline website/docusaurus.config.js]: Change this please

File: .github/workflows/deploy-to-github-pages.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+          npm run build
+
+      - name: Deploy to GitHub Pages
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          # Build output to publish to the `gh-pages` branch:
+          publish_dir: ./website/build
+          user_name: github-actions[bot]
+          user_email: 41898282+github-actions[bot]@users.noreply.github.com
```
Comment (dgreene1) [inline .github/workflows/deploy-to-github-pages.yml:33]: @yhy-1 who’s anonymized email is this?

File: .github/workflows/deploy-to-github-pages.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      - name: Install dependencies
+        run: npm ci
+      - name: Build website
+        working-directory: ./website
+        run: |
+          npm i
+          npm run build
+
+      - name: Deploy to GitHub Pages
+        uses: peaceiris/actions-gh-pages@v3
```
Comment (dgreene1) [inline .github/workflows/deploy-to-github-pages.yml:27]: I’m surprised that you have to use a 3rd party command when docusaurus provides their own command. Did you try this?

> We provide a docusaurus deploy command that helps you deploy your site from the source branch to the deployment branch in one command: clone, build, and commit.

File: .github/workflows/deploy-to-github-pages.yml (hunk 3 above)
Comment (yhy-1) [reply .github/workflows/deploy-to-github-pages.yml:33]: This was part of Docusaru's documentation.   https://docusaurus.io/docs/deployment#triggering-deployment-with-github-actions. It looks like https://github.com/actions/checkout/discussions/479#discussioncomment-625461 it shows GitHub action bot for the commit. Not sure the number is needed but it was in the documentation.

File: .github/workflows/deploy-to-github-pages.yml (hunk 4 above)
Comment (yhy-1) [reply .github/workflows/deploy-to-github-pages.yml:27]: This was part of Docusaru's documentation.   https://docusaurus.io/docs/deployment#triggering-deployment-with-github-actions.

Comment (yhy-1) [reply .github/workflows/deploy-to-github-pages.yml:27]: This is what was said in the documentation. The original deploy workflow will always be listed as skipped in the PR check suite status, which is not communicative of the actual status and provides no value to the review process. We, therefore, propose to manage them as separate workflows instead.

We will use a popular third-party deployment action: [peaceiris/actions-gh-pages](https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-docusaurus).

File: .github/workflows/deploy-to-github-pages.yml (hunk 3 above)
Comment (dgreene1) [reply .github/workflows/deploy-to-github-pages.yml:33]: K that works for me.

File: .github/workflows/deploy-to-github-pages.yml (hunk 4 above)
Comment (dgreene1) [reply .github/workflows/deploy-to-github-pages.yml:27]: Works for me

**Comment by yhy-1** [comment]
image
@dgreene1  I tried using the deploy command but it does not recognize GitHub action for  GIT_USER and GIT_PASS. I don't think we want to set our email and password. Third-party looks like a better option.

**Review by dgreene1 — APPROVED** [review]

**Pull request merged**